api/proto/teleport/legacy/types/types.proto

```diff
@@ -2964,6 +2964,37 @@ message RoleConditions {
   // GroupLabelsExpression is a predicate expression used to allow/deny
   // access to user groups.
   string GroupLabelsExpression = 37 [(gogoproto.jsontag) = "group_labels_expression,omitempty"];
+  // SPIFFE is used to allow or deny access to a role holder to generating a
+  // SPIFFE SVID.
+  repeated SPIFFERoleConditions SPIFFE = 38 [(gogoproto.jsontag) = "spiffe,omitempty"];
+}
+
+// SPIFFERoleConditions sets out which SPIFFE identities this role is allowed or
+// denied to generate. The Path matcher is required, and is evaluated first. If,
+// the Path does not match then the other matcher fields are not evaluated.
+message SPIFFERoleConditions {
+  // Path specifies a globby style matcher for the SPIFFE ID path. It should not
+  // include the trust domain and should start with a leading slash.
+  // Matching:
+  // - '*' matches zero or more characters.
```

Comment:
Do we need more advanced matching e.g '*' stops at slash, and '**' ignores slashes?

Reply:
Going to address this in a future PR for the RBAC in particular.